++ dirname ./startserver.sh
+ BASEDIR=.
+ SCRIPTPATH=./../../rtun-routing
+ export PYTHONPATH=./../../torpy-rtun-fork/
+ PYTHONPATH=./../../torpy-rtun-fork/
+ python3 ./../../rtun-routing/rtun.py -v ERROR -g BM10 -f rendezvous.txt -t peer1peer2 -n default -i 3 -d 2
Exception in thread Thread-959:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.9/threading.py", line 1266, in run
    self.function(*self.args, **self.kwargs)
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 573, in runDijkstra
    for n , w in args[0][current_vertex].items():
KeyError: 'P3'
Exception in thread HEART BEAT:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 706, in run
    self.broadcastHB()
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 718, in broadcastHB
    send_to_stream( neighbour['NID'], message)
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 953, in send_to_stream
    rendezvous.snd_data(message, 
  File "/home/vboxuser/Git/imt4905/rtun-routing/rendezvous.py", line 342, in snd_data
    rcv_sock.send_cell(relay_cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 88, in send_cell
    self._socket.write(buffer)
  File "/usr/lib/python3.9/ssl.py", line 1118, in write
    return self._sslobj.write(data)
BrokenPipeError: [Errno 32] Broken pipe
Exception in thread SENDER:
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 954, in _bootstrap_inner
    self.run()
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 672, in run
    self.clientSide()
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 691, in clientSide
    send_to_stream(dict['NID'], message)
  File "/home/vboxuser/Git/imt4905/rtun-routing/lsr.py", line 953, in send_to_stream
    rendezvous.snd_data(message, 
  File "/home/vboxuser/Git/imt4905/rtun-routing/rendezvous.py", line 342, in snd_data
    rcv_sock.send_cell(relay_cell)
  File "/home/vboxuser/Git/imt4905/torpy-rtun-fork/torpy/cell_socket.py", line 88, in send_cell
    self._socket.write(buffer)
  File "/usr/lib/python3.9/ssl.py", line 1118, in write
    return self._sslobj.write(data)
BrokenPipeError: [Errno 32] Broken pipe
Exception ignored in: <module 'threading' from '/usr/lib/python3.9/threading.py'>
Traceback (most recent call last):
  File "/usr/lib/python3.9/threading.py", line 1428, in _shutdown
    lock.acquire()
KeyboardInterrupt: 
